games-engines/scummvm/scummvm-1.8.1.recipe

```diff
@@ -0,0 +1,95 @@
+DESCRIPTION="ScummVM is a program which allows you to run certain classic \
+graphical point-and-click adventure games, provided you already have their \
+data files. The clever part about this: ScummVM just replaces the executables \
+shipped with the games, allowing you to play them on systems for which they \
+were never designed!"
+HOMEPAGE="http://scummvm.org"
+COPYRIGHT="2001-2014 ScummVM Team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="http://scummvm.org/frs/scummvm/$portVersion/scummvm-$portVersion.tar.xz"
+CHECKSUM_SHA256="935f53d7e4c460fbc22c58fee5007d4858891e4961831f53f575a3594b7f612a"
+ADDITIONAL_FILES="scummvm.rdef"
+
+ARCHITECTURES="!x86_gcc2 x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	scummvm$secondaryArchSuffix = $portVersion
+	app:scummvm$secondaryArchSuffix = $portVersion
+"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libfaad$secondaryArchSuffix
+	lib:libflac$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libfluidsynth$secondaryArchSuffix
+	lib:libgl$secondaryArchSuffix
+	lib:libglu$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libmad$secondaryArchSuffix
+	lib:libmpeg2$secondaryArchSuffix
+	lib:libogg$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libSDL$secondaryArchSuffix
+	lib:libtheora$secondaryArchSuffix
+	lib:libvorbis$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:libfaad$secondaryArchSuffix
+	devel:libflac$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libfluidsynth$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libglu$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libmad$secondaryArchSuffix
+	devel:libmpeg2$secondaryArchSuffix
+	devel:libogg$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libSDL$secondaryArchSuffix
+	devel:libtheora$secondaryArchSuffix
+	devel:libvorbis$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+"
+
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:git
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:nasm
+	cmd:pkg_config$secondaryArchSuffix
+"
+
+BUILD()
+{
+	libtoolize --force --copy --install
+	CPPFLAGS=`freetype-config --cflags` ./configure --prefix=$prefix \
+	--bindir=$appsDir --libdir=$libDir --datarootdir=$dataDir \
+	--mandir=$manDir --docdir=$docDir \
+	--disable-debug --enable-release #--disable-all-engines
+	#--disable-all-engines can be used to only compile scummvm without
+	#engines, which makes it build faster, those that are still under
+	#construction can be built with --enable-all-engines for testing purposes
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	mv $appsDir/scummvm $appsDir/ScummVM
+	addResourcesToBinaries $portDir/additional-files/scummvm.rdef \
+		$appsDir/ScummVM
+
+	addAppDeskbarSymlink $appsDir/ScummVM ScummVM
+}
```

Comment:
If **`grep "^check:" work-x86-1.8.1-1/source/scumm-1.8.1/Makefile`** yelds something then you can (and probably should) add this at the end of the recipe:
```Shell

TEST()
{
	make check
}
```

With the TEST() function you can launch the test (after it has been built) by calling **`haikuporter --test scummvm_x86-1.8.1`** on x86_gcc2 or **`haikuporter --test scummvm-1.8.1`** on x86 or x86_64.
(I don't know if you have the infinite loop bug from time to time, but if haikuporter loops with **`waiting for build package scummvm_x86-1.8.1-1 to be activated`** , then you just need to press Control-C and retry.)